### PR TITLE
spdx: update to SPDX license list version: 3.11 2020-11-25

### PR DESCRIPTION
**Pull request by bboozzoo**
Some licenses were missing. Update to the latest version of the SPDX license
list.

Fixes: https://bugs.launchpad.net/snapcraft/+bug/1903609

@pedronis iirc you mentioned that the store uses the `spdx` package from snapd, can you take a look whether this can break anything on the store side?

### CODE REVIEW

File: spdx/licenses.go [hunk 1]
```diff
@@ -19,193 +19,226 @@
 
 package spdx
 
-// from https://spdx.org/licenses/
+// From https://spdx.org/licenses/
+// Version: 3.10 2020-08-03
+//
+// jq < json/licenses.json '.licenses | .[] | select(.isOsiApproved == true) | .licenseId' | sort | sed -e 's/$/,/
```
Comment (zyga) [inline spdx/licenses.go:25]: Nice touch :)

**Review by zyga — APPROVED** [review]
+1

**Review by mvo5 — APPROVED** [review]
Nice

**Comment by pedronis** [comment]
marking blocked because we need to discuss the timing around enabling/announcing this

File: spdx/licenses.go [hunk 2]
```diff
@@ -19,193 +19,227 @@
 
 package spdx
 
-// from https://spdx.org/licenses/
+// From https://spdx.org/licenses/
+// data files: https://github.com/spdx/license-list-data
+// Version: 3.10 2020-08-03
+//
+// jq < json/licenses.json '.licenses | .[] | select(.isOsiApproved == true) | .licenseId' | sort | sed -e 's/$/,/'
```
Comment (pedronis) [inline spdx/licenses.go:26]: thanks for the sorting

File: spdx/parser.go [hunk 3]
```diff
@@ -40,9 +39,6 @@ type licenseID string
 
 func newLicenseID(s string) (licenseID, error) {
 	needle := s
-	if strings.HasSuffix(s, "+") {
-		needle = s[:len(s)-1]
-	}
```
Comment (pedronis) [inline spdx/parser.go:45]: what is this change about? there's no matching test change either?

**Review by pedronis — COMMENTED** [review]
did a quick pass

File: spdx/parser.go (hunk 3 above)
Comment (bboozzoo) [reply spdx/parser.go:45]: I'll push a test update for that. With this block, a license such as `GPL-2.0++` would be considered valid.

**Review by pedronis — APPROVED** [review]
thanks

**Comment by mvo5** [comment]
Given that this is updated, no longer blocked, has 2 plus one and should go into 2.49 - I think this can be merged now?

**Comment by bboozzoo** [comment]
Can we land it this week then?

**Comment by pedronis** [comment]
> Can we land it this week then?

yes, think so

**Pull request merged**